lib/writer.js

```diff
@@ -204,14 +210,21 @@ Writer.prototype.connectTo = function(addr){
  */
 
 Writer.prototype.close = function(fn){
-  debug('start close');
+  debug('close');
+
+  if (fn) this.once('close', fn);
+
+  this.init.each(function(conn){
+    conn.destroy();
+  });
+
+  if (0 === this.conns.size()) {
+    this.emit('close');
+    return;
+  }
 
-  var n = this.conns.size();
   this.conns.each(function(conn){
-    conn.end(function() {
-      debug('%s - conn ended', conn.addr);
-      if (fn) --n || fn();
-    });
+    conn.close();
```

Comment:
Since this will emit a `close` event and you've added a `.once('close', fn)`, the writer will call the callback function as soon as the first connection is closed instead of waiting for all connections to close, as the original method does.